src/util/redux.js

```diff
@@ -0,0 +1,235 @@
+import _ from 'lodash';
+import { createSelector } from 'reselect';
+
+/**
+ * thunk
+ *
+ * Marks a function on the reducer tree as a thunk action creator so it doesn't
+ * get incorporated into the redux reducer
+ *
+ * @return {function} with `isThunk` set to `true`
+ */
+export function thunk(fn) {
+	fn.isThunk = true;
+	return fn;
+}
+
+/**
+ * getReduxPrimitives
+ *
+ * Creates a redux reducer and connectors (inputs to redux-react's `connect`)
+ *
+ * @param {Object} param
+ * @param {Object} param.initialState - the initial state object that the reducer will return
+ * @param {Object} param.reducers - a tree of lucid reducers
+ * @param {string[]} param.rootPath - array of strings representing the path to local state in global state
+ * @param {Object} param.selectors - a tree of lucid selectors
+ * @return {Object} redux reducer and connectors
+ */
+
+export function getReduxPrimitives({
+	initialState,
+	reducers,
+	rootPath = [],
+	rootSelector = _.identity,
+	selectors
+}) {
+
+	// we need this in scope so actionCreators can refer to it
+	let dispatchTree;
+
+	const reducer = createReduxReducer(reducers, initialState, rootPath);
+	const selector = selectors ? reduceSelectors(selectors) : _.identity;
+	const rootPathSelector = state => _.isEmpty(rootPath) ? state : _.get(state, rootPath);
+	const mapStateToProps = createSelector(
+		[rootPathSelector],
+		rootState => rootSelector(selector(rootState))
+	);
+	const mapDispatchToProps = dispatch => getDispatchTree(reducers, rootPath, dispatch);
+
+	return {
+		reducer,
+		connectors: [
+			mapStateToProps,
+			mapDispatchToProps,
+			mergeProps
+		]
+	};
+
+	/**
+	 * createActionCreator
+	 *
+	 * @param {function} node - a node in the the reducer tree, either a reducer or a thunk
+	 * @param {string[]} path - the path to the reducer in the reducer tree
+	 * @param {string[]} rootPath - array of strings representing the path to local state in global state
+	 * @return {function} action creator that returns either an action or a thunk
+	 */
+	function createActionCreator(node, path, rootPath) {
+		if (node.isThunk) {
+			return function thunk(...args) {
+				return function thunkInner(dispatch, getState) {
+					const localPath = _.slice(path, rootPath.length, -1);
+					const localDispatchTree = _.isEmpty(localPath) ? dispatchTree : _.get(dispatchTree, localPath);
+					const getLocalState = _.isEmpty(rootPath) ? getState : () => _.get(getState(), _.slice(path, 0, -1));
```

Comment:
i think it's more readable to use `_.dropRight(path, 1)` or `_.take(path, path.length - 1)` instead of the `slice`